Documentation/ceph-best-practices.md

```diff
@@ -0,0 +1,516 @@
+> `requiredDuringSchedulingIgnoredDuringExecution` Pod Anti-affinity rule.
+
+### Ceph OSDs
+
+There is a lot of planning that goes into the placement of monitors, and this is also true for OSDs.
+Fortunately, because the planning is already done with the monitors and because we have discussed
+the methods, it is quite a bit easier to discuss best practices for the OSDs.
+
+![Example Datacenter - OSD placement](media/ceph-bp-example-datacenter-osd-placement.png)
+
+There are two ways to select nodes to use for OSDs:
+* Apply Kubernetes Node labels and tell Rook to look for those labels. Specify in the cluster.yaml
+  `storage:useAllNodes: true` and specify `osd` `nodeAffinity` using a `ceph-osd=true` label using
+  the same Affinity methods we used for `mon`s.
+ * Specify node names in the `CephCluster` definition (`cluster.yaml`) individually in `storage:nodes`.
+
+Choosing which option to use depends on your desired management strategy. There is no single
+strategy we would recommend over any other.
+
+### Other Ceph Daemons
+
+Placing the other Ceph daemons follows the same logic and methods as MONs and OSDs: MGR, MDS, RGW,
+NFS-Ganesha, and RBD mirroring daemons can all be placed as desired. We won't discuss these in
+details here since there are fewer strong recommendations for where these should be placed. For more
+information, see the [placement configuration settings](./ceph-cluster-crd.md#placement-configuration-settings].)
+
+
+## Hardware Requirements and Requests
+
+Kubernetes can watch the system resources available on nodes and can help schedule applications
+automatically. Kubernetes uses Resource Requests to do this. For Rook, we are notably concerned
+about Kubernetes' scheduling of Ceph daemons.
+
+Kubernetes has two Resource Request types: Requests and Limits. Requests govern scheduling, and
+Limits instruct Kubernetes to kill and restart application Pods when they are over-consuming given
+Limits.
+
+When there are Ceph hardware requirements, treat those requirements as Requests, not Limits. This is
+because all Ceph daemons are critical for storage, and it is best to never set Resource Limits for
+Ceph Pods. If Ceph Daemons are over-consuming Requests, there is likely a failure scenario
+happening. In a failure scenario, killing a daemon beyond a Limit is likely to make an already bad
+situation worse. This could create a “thundering herds” situation where failures synchronize and
+magnify.
+
+Generally, storage is given minimum resource guarantees, and other applications should be limited so
+as not to interfere. This guideline already applies to bare-metal storage deployments, not only for
+Kubernetes.
+
+As you read on, it is important to note that all recommendations can be affected by how Ceph daemons
+are configured. For example, any configuration regarding caching. Keep in mind that individual
+configurations are out of scope for this document.
+
+### Resource Requests - MON/MGR
```

Comment:
We have the min resource limits documented in ceph-cluster-crd.md. How about linking to that?